### ISSUE TITLE
CQ500 dataset to TFRecord

### ISSUE THREAD

**Issue by niladell** [niladell/unsupervised-medical-learning#1]
@pjaweh in order to fit in with the approach we discussed we have to convert our current dataset to TFRecord format.  As a reference on how to do that check the [CIFAR to TFRecord](https://github.com/niladell/unsupervised-medical-learning/blob/master/src/util/generate_cifar10_tfrecords.py) file. I'm not completely sure about the complete list of things that should be saved, but at least the image data (obviously), the subject ID, and the z-axis (slice height) value.

As constraints on that, I'd say that 

- the data in the TFRecord has to be encoded in bytes as it is in the example file (i.e avoid encoding strings as `'image': _bytes_feature(img.tostring())`), and I think
- one _tfrecord_ file should be created per image (slice), something like [this](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#list_files) could be helpful for instance. 

However, that may be just one way, so if you were to feel more comfortable with something different (like creating a set of .h5py files and generating samples from there) that'd be okay as long as it's TPU compatible and not significantly more inefficient than TFRecord (e.g. loading all into RAM).

**Comment by niladell** [comment]
@pjaweh did you make a script to run the code for all the files? Can we close this?

**Comment by niladell** [comment]
So I changed a bit the [dcm_to_tfRecord](https://github.com/niladell/unsupervised-medical-learning/blob/cq500_input_functions/src/datamanager/convert_dcm_to_unique_tfRecord.py) file and posted a new version. Basically two main things. 

1.  One is that it may be important for us to have a single file (fewer calls to the bucket).

2. Is related to the type. The previous way we had to do that basically passed the file like that but there was no way to reconstruct it (in a normal manner), so it's best to use pydicom to reconstruct a numpi array and then work from there. 
The issue we have when doing so is that we get an array of *int16*, meaning that the value can be in the range -+32767. There is nothing in the negative part except for -2000 which indicates that that pixel is not part of the scan (right?), just a filler. However, on the positive (>0) part, the maximum values range a lot. With *unit8* we just normalized by 255 (max possible value). However among all the dcms that @InezPereira has in the _pca_experimets_ folder the maximum value was 5067 and in general there was a high variability on the maximum value of the dcms. So, at this point, I'm not sure what to do; for sure dividing by 32767 is not the way to go. Our best bet may be to normalize between 0 and 1 based on the maximum of each *dcm*, and that's it. But that's where I'm not sure of how a higher or lower intensity could be important or if we just care about the relative value (fortunately I think the norm is the latter but I'm not sure). 
As a last note, about the filler values (-2000) I guess we can use that to create a mask and not care about the output there (and avoid backpropagating stuff there plus helping on the training)... so another TODO more on the list...

In any case this is the part I was talking about:

https://github.com/niladell/unsupervised-medical-learning/blob/245628f7e7160471851442d732b2e4828e45d1d1/src/datamanager/convert_dcm_to_unique_tfRecord.py#L30-L38

What do you guys think?
